rfcs/0000-ipfs-net.md

```diff
@@ -0,0 +1,67 @@
+## Summary
+[summary]: #summary
+
+Use ipfs as a content sharing network for history sync. There is also space to use their "gossipsub" pubsub protocol (when it's finished) for efficient broadcasting.
+
+## Motivation
+
+[motivation]: #motivation
+
+As part of the new transition frontier controller ([RFC #8](./0008-transition-frontier-controller.md)), we need to implement "history sync", downloading parts of the history from peers who were online to see it. This is essentially a distributed lookup of breadcrumb by  We currently use a naive algorithm for a similar problem, but it has known limitations. libp2p is the networking code that ipfs uses, which handles things like network transport concerns, NAT traversal, and a DHT. On top of that, ipfs builds a distributed content-addressed data store.
+
+## Detailed design
+
+[detailed-design]: #detailed-design
+
+ipfs has an HTTP API. The signature we expose for using it is:
+
+```ocaml
+module type Block_storage_intf = sig
+  type t
+
+  val create : peers:Peer.t list -> state_dir:string -> listen:Host_and_port.t -> t Deferred.Or_error.t
+  (** Start the helper process, listening on external address [listen], using [peers] to bootstrap and storing state in [state_dir]. *)
+
+  val shutdown : t -> unit
+  (** Shutdown the helper process *)
+
+  val provide : data:Bigstring.t -> string Deferred.Or_error.t
+  (** Offer a piece of data to the network, giving back a handle that can be used with [fetch].
+
+      This will persist the data in local storage until [remove] is called.
+   *)
+
+  val remove : handle:string -> unit Deferred.Or_error.t
+  (** Remove the data referred to by [handle] from local storage and stop offering it to the network. *)
+
+  val fetch : handle:string -> Bigstring.t Deferred.Or_error.t
+  (** Attempt to retrieve a piece of data. *)
+end
+```
+
+Internally, these are implemented using ipfs's `block` API. The three main methods correspond exactly to `block/put`, `block/rm`, and `block/get` respectively. We will spwan and manage our own ipfs daemon that isn't necessarily connected to the wider ipfs network (by not using the default bootstrap list). The provided blocks persist past restarts - the frontier will have to keep track of which block it stops caring about and explicitly `remove` them.
```

Comment:
It would be helpful to discuss the technical details of a `block` for IPFS. Also, include a  link of the doc to `block`

Reply:
Does the new text satisfy you? It's not much, but blocks aren't very complicated. Any more details would require diving pretty deep into ipfs minutiae.